File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int counter = 0;
        int number = -1;
        for(int num: nums){
            if(counter == 0){
                counter = 1;
                number = num;
            } else if (number == num){
                counter += 1;
            } else {
                counter -= 1;
            }
        }
        return number;
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        counter = {}
        for num in nums:
            if num not in counter:
                counter[num] = 1
            else:
                counter[num] = counter[num] + 1

        length = len(nums)
        for key, val in counter.items():
            if val > length / 2:
                return key
        return -1